### PR TITLE
Pager augmenetation

### PR DESCRIPTION
**Pull request by boki1**
# Description
Refactor the Pager module. Make components based on `Policy` (e.g `StackAllocator` for `Page` allocation), which improves the modularity of the design and enables the development of the VL entry component.

# Test suite
```
. tools/env.sh
eu config && eu test +runtest
```

# Environment
Using gcc (GCC) 11.1.0 on a x86_64 machine with GNU/Linux.

--------------------

#### This PR makes changes which ...

- [x] are thoroughly tested
- [x] are well documented

### CODE REVIEW

**Comment by StoyanTinchev** [comment]
conversion from 'internal::storage::Position' (aka 'unsigned long') to signed type 'std::vector<int>::value_type' (aka 'int')

```
			b.push_back(Position(dist128(rng)));
```
								
[src/core/storage/btree/Node.cpp](https://github.com/boki1/eugene/blob/54cc673081ec08d57cf824513b86734707cf2516/src/core/storage/btree/Node.cpp#L38) (Line 38)

**Comment by StoyanTinchev** [comment]
Here, 144 and 138 rows are narrowing conversion from `uint32_t` to signed type `std::vector<int>` (aka `int`)

```
		l.m_vals.push_back(i);
```
								
[src/core/storage/btree/Node.cpp](https://github.com/boki1/eugene/blob/54cc673081ec08d57cf824513b86734707cf2516/src/core/storage/btree/Node.cpp#L145) (Line 145)

File: src/core/storage/Pager.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#include <algorithm>
```
Comment (StoyanTinchev) [inline src/core/storage/Pager.cpp:1]: unused

File: src/core/storage/Pager.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#include <fstream>
+#include <array>
+#include <filesystem>
```
Comment (StoyanTinchev) [inline src/core/storage/Pager.cpp:5]: unused

**Review by StoyanTinchev — CHANGES_REQUESTED** [review]
Please consider the comments that I left (aka. good job)

File: src/core/storage/Pager.h [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+	}
+
+	void free(const Position) {
+		throw BadAlloc{};
+	}
+
+	/// If the page is below the cursor, it is considered as allocated since there is no "official" way of freeing
+	/// pages using this allocator.
+	/// Note: The implementation does track whether the pos is actually identifing a valid page.
+	constexpr bool has_allocated(const Position pos) const noexcept {
```
Comment (StoyanTinchev) [inline src/core/storage/Pager.h:102]: Can be marked [[nodiscard]]

**Review by StoyanTinchev — APPROVED** [review]
Just declare that function with [[nodiscard]]. Overall - good job!

File: src/core/storage/Pager.h (hunk 3 above)
Comment (boki1) [reply src/core/storage/Pager.h:102]: Yup.

**Pull request merged**